dataplane/vppagent/pkg/converter/kernel_connection_converter.go

```diff
@@ -110,7 +111,7 @@ func (c *KernelConnectionConverter) ToDataRequest(rv *rpc.DataRequest, connect b
 			IpAddresses: ipAddresses,
 			HostIfName:  tmpIface,
 			Veth: &linux_interfaces.LinuxInterfaces_Interface_Veth{
-				PeerIfName: m.GetParameters()[connection.InterfaceNameKey],
+				PeerIfName: c.conversionParameters.Name,
```

Comment:
@haiodo So this was the root issue?  It looks like the rest of the chain is adding testing (which is good!).

Reply:
Yes, this is root cause for GKE for not working at all, but still we had #855 and it still not working in 50% of times.